src/DistributedMetadata/CatalogService.cpp

```diff
@@ -378,6 +379,25 @@ std::pair<bool, bool> CatalogService::columnExists(const String & database, cons
     return {true, false};
 }
 
+String CatalogService::getColumnType(const String & database, const String & table, const String & column) const
+{
+    const auto & tables = findTableByName(database, table);
+    const auto & query_ptr = parseQuery(tables[0]->create_table_query, global_context);
+    const auto & create = query_ptr->as<const ASTCreateQuery &>();
+    const auto & columns_ast = create.columns_list->columns;
+
+    for (auto ast_it = columns_ast->children.begin(); ast_it != columns_ast->children.end(); ++ast_it)
+    {
+        const auto & col_decl = (*ast_it)->as<ASTColumnDeclaration &>();
+        if (col_decl.name == column)
+        {
+            return queryToString(col_decl.type);
+        }
+    }
+
+    return "";
```

Comment:
do we have a case column type doesn't exist ?

Reply:
fixed